src/backend.md

```diff
@@ -173,9 +173,28 @@ For serverless projects - method 3 is preferred, but not always. When you have a
 
 ### Transactions
 
-1. Avoid initiating read network calls within a transaction before the first SQL statement.
+One of the key principles in effective transaction management is the avoidance of prolonged transaction durations, particularly with regards to networking operations. This precaution is essential because database transactions involve locking the database, potentially resulting in decreased database performance and an increased likelihood of encountering deadlocks.
+
+1. As a general rule, it's advisable to abstain from incorporating SELECT statements within a transaction prior to the first SQL statement that explicitly begins the transaction. Utilize such SELECTs only if it's imperative to guarantee that the state of the SELECT statement remains unaltered before the transaction commencement.
+
+   ```typescript
+   // bad
+   return sequelize.transaction(async (transaction) => {
+     const { id: userId } = await User.findOne({ email }, { transaction });
+     await changeUserBalance(userId, amount, { transaction });
+     await cashBackUser(userId, amount, { transaction });
+   });
 
-   One of the key principles in effective transaction management is the avoidance of prolonged transaction durations, particularly with regards to networking operations. This precaution is essential because database transactions involve locking the database, potentially resulting in decreased database performance and an increased likelihood of encountering deadlocks.
+   // good
+   const { id: userId } = await User.findOne({ email });
+
+   return sequelize.transaction(async (transaction) => {
+     await changeUserBalance(userId, amount, { transaction });
```

Comment:
Do we assume a SELECT FOR UPDATE inside?

Reply:
inside yes, otherwise it makes no sense?

Reply:
Oh, it is "change", I read it as "check" for the first time. It makes sense even without "SELECT FOR UPDATE" since there will be "UPDATE" inside, and UPDATE will lock the row until COMMIT.